### ISSUE TITLE
vsg does not build Markdown files in subdirectories

### ISSUE THREAD

**Issue by vktec** [vktec/vsg#5]
## Why wasn't this noticed?

Because it builds `index.md` in subdirectories just fine, and I didn't clean out the output directory when testing.

## What's causing it?

When building subdirectories, vsg adds `Page` objects to a list, which is then used as the `children` property of the index page. However, the `build` function does not traverse the tree of pages, and therefore only renders toplevel pages, which happens to include index pages of subdirectories.

In other words, the pages are getting rendered, but not saved.

## What's the fix?

Get the `build` function to traverse the tree.

**Comment by vktec** [comment]
I ended up splitting the `Page` saving section of `build` out into `save_pages`, allowing me to use recursion.
